doc: reformat and improve overview.rst (#6343)
ocaml/dune: doc/overview.rst

Introduction
============

Dune is a build system for OCaml (with support for Reason and Coq). It is not
intended as a completely generic build system that's able to build any project
in any language. On the contrary, it makes lots of choices in order to encourage
a consistent development style.

This scheme is inspired from the one used inside Jane Street and adapted to the
opam world. It has matured over a long time and is used daily by hundreds of
developers, which means that it is highly tested and productive.

When using Dune, you give very little, high-level information to the build
system, which in turn takes care of all the low-level details from the
compilation of your libraries, executables, and documentation to the
installation, setting up of tests, and setting up development tools such as
Merlin, etc.

In addition to the normal features expected from an OCaml build system, Dune
provides a few additional ones that separate it from the crowd:

-  You never need to tell Dune the location of things such as libraries. Dune
   will discover them automatically. In particular, this means that when you

>    want to re-organize your project, you need nothing other than to rename your

Reviewer: ```suggestion
   want to reorganise your project, you need nothing other than to rename your
```
Removed hyphen and changed to British spelling (which we have in our Style Guide, but the Dune docs might be American spelling. Do you know?)

Author: I don't know if we take British or American English as convention. cc @rgrinberg do you know?

Reviewer 2: I'd stay just stick what's already in use in dune. I don't have a preference, but let's keep it consistent.

   directories, Dune will do the rest.

-  Things always work the same whether your dependencies are local or installed
   on the system. In particular, this means that you can insert the source for a
   project dependency in your working copy, and Dune will start using it
   immediately. This makes Dune a great choice for multi-project development.

-  Cross-platform: as long as your code is portable, Dune will be able to
   cross-compile it. Read more in the :ref:`cross-compilation` section.

-  Release directly from any revision: Dune needs no setup stage. To release

>    your project, simply point to a specific Git tag (named revision. Of course,

Reviewer: ```suggestion
   your project, simply point to a specific Git tag (named revision). Of course,
```

interesting things are: libraries, executables, tests, documents to install,
etc.

> We recommended organizing your project to have exactly one library per

Reviewer: ```suggestion
We recommended organising your project to have exactly one library per
```
Same with the British spelling. (I think in previous docs we did this, so if it should be American, I'll make a note of it.)
